lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -245,6 +273,53 @@ def add_authorization(doc, authorization)
         doc.send('transaction-id', authorization)
       end
 
+      def add_fraud_info(doc, options)
+        doc.send('transaction-fraud-info') do
+          doc.send('shopper-ip-address', options[:ip]) if options[:ip]
+        end
+      end
+
+      def add_alt_transaction_purchase(doc, money, payment_method_details, options)
+        doc.send('merchant-transaction-id', truncate(options[:order_id], 50)) if options[:order_id]
+        doc.send('soft-descriptor', options[:soft_descriptor]) if options[:soft_descriptor]
+        add_amount(doc, money, options)
+
+        vaulted_shopper_id = payment_method_details.vaulted_shopper_id
+        doc.send('vaulted-shopper-id', vaulted_shopper_id) if vaulted_shopper_id
+
+        if payment_method_details.check?
+          add_echeck_transaction(doc, payment_method_details.payment_method, options, vaulted_shopper_id.present?)
+        end
+
+        add_fraud_info(doc, options)
+        add_description(doc, options)
+      end
+
+      def add_echeck_transaction(doc, check, options, vaulted_shopper)
+        unless vaulted_shopper
+          doc.send('payer-info') do
+            add_personal_info(doc, check, options)
+            add_echeck_company(doc, check)
+          end
+        end
+
+        doc.send('ecp-transaction') do
+          add_echeck(doc, check) unless vaulted_shopper
+        end
+
+        doc.send('authorized-by-shopper', options[:authorized_by_shopper])
```

Comment:
This field must always be `true` for ACH/ECP and other alternate payments to be accepted. However, BlueSnap requires that the merchant have a checkbox with specific verbiage on their checkout page and require the customer to agree to terms. 

I left it up to our customers to pass this value in to ensure they were meeting the gateway's requirements.